Kinvey/Kinvey/User.swift

```diff
@@ -996,10 +996,34 @@ open class User: NSObject, Credential, Mappable {
     }
     
     /// Sign out the current active user.
-    open func logout() {
-        if self == client.activeUser {
-            client.activeUser = nil
+    @discardableResult
+    open func logout(
+        options: Options? = nil,
+        completionHandler: ((Result<Void, Swift.Error>) -> Void)? = nil
+    ) -> Request {
+        let request = client.networkRequestFactory.buildUserLogout(
+            user: self,
+            options: options
+        )
+        Promise<Void> { fulfill, reject in
+            request.execute { data, response, error in
+                if let response = response,
+                    response.isOK
+                {
+                    fulfill()
+                } else {
+                    reject(error ?? buildError(data, response, error, self.client))
```

Comment:
@heyzooi two suggestions -
1 - let's bring this up in standup and check what we're doing on the other platforms for logout failures... does an error in the network operation get reported as an error, or is it still a success (since we logged the user out locally)?
2 - please make sure the docs describe the failure case properly.

Reply:
@tejasranade as discussed in our standup, the current implementation is the expected behaviour